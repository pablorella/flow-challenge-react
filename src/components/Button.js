import React from 'react';
import styled from 'styled-components';

const StyledButton = styled.button`
  border-radius: 5px;
  background: ${({ disabled }) => (disabled ? '#777' : '#2ECCFA')};
  margin-top: 30px;
  color: #fff;
  font-weight: bold;
  font-size: 1.2rem;
  cursor: pointer;
`

const Button = ({title, disabled, handleOnClick}) => {
  return <StyledButton disabled={disabled} type="button" onClick={handleOnClick} >{title}</StyledButton>
}

export default Button;