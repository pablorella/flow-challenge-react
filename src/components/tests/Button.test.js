import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import Button from '../Button';


describe('Button', () => {

  afterEach(cleanup);

  it('should matches snapshot', () => {
    const { asFragment } = render(<Button />)
    expect(asFragment()).toMatchSnapshot();
  })


  it('should render Button with the correct title', () => {
    const title = 'test title';
    const { getByText } = render(<Button title={title} />)

    expect(getByText(title)).toBeTruthy();

  })

  it('should render Button as disabled', () => {
    const disabled = true;
    const { getByRole } = render(<Button disabled={disabled} />)
    expect(getByRole('button')).toHaveAttribute('disabled')
  })


  it('should call the correct action when is clicked', () => {

    const onClick = jest.fn();

    const { getByRole } = render(<Button type="button" disabled={false} title="test title" handleOnClick={onClick} />)
    fireEvent.click(getByRole('button'));

    expect(onClick).toHaveBeenCalled()
  })
})