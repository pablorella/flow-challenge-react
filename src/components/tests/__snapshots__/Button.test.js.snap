// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-AxjAm eGcxxL"
    type="button"
  />
</DocumentFragment>
`;

exports[`Button should render Button with the correct title 1`] = `
<DocumentFragment>
  <button
    class="sc-AxjAm eGcxxL"
    type="button"
  >
    test title
  </button>
</DocumentFragment>
`;

exports[`Button should rmatches snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-AxjAm eGcxxL"
    type="button"
  />
</DocumentFragment>
`;
